cps/templates/layout.html

```diff
@@ -226,6 +226,11 @@ <h4 class="modal-title" id="bookDetailsModalLabel">{{_('Book Details')}}</h4>
             $("#btn-upload").change(function() {
                 $("#form-upload").submit();
             });
+            $(document).ready(function() {
+              var inp = $('#query').first()
+              var val = inp.val()
+              inp.val('').blur().focus().val(val)
+            })
```

Comment:
Hi @celogeek - thanks very much for your contributions to this project! I use it extensively at home. :) 

Regarding this change which was merged to master about a month ago, would you mind walking me through why the value was cleared, then focus on the search box was forced on the user and then the value re-set? I don't understand why it should clear and re-set the value and why focus? It feels a bit jarring to the user does it not?

Reply:
Looks like suggested method 3 to get the cursor to the end of a focused text field:
https://subinsb.com/move-cursor-to-end-input-textarea-jquery/

Reply:
Yeah, I pick the trick from somewhere on the web.
The step to be able to have a field focus and be at the end of the current text is:

- clear
- blur (unfocus)
- focus
- set back the value

This way you are focus on the field, and your cursor is placed at the end of the input text.

Reply:
Thanks very much for explaining. Would it be a good idea to only do this if the value is not blank? Given that this logic is run on every page regardless of whether or not there is a search term present, it causes mobile keyboards to jump up while the user is scrolling which can be quite jarring. When I noticed it start happening recently I thought I had somehow broken something in my library. Maybe something along the lines of: 

`if (val.trim()) {
  inp.val('').blur().focus().val(val);
}`

Just a thought - I am sorted for now as I've just commented out that code but it might be a good idea for the next PR. 

Thanks for being patient as well - this is the first time I've done anything other than lurk on any code repos :)

Reply:
I have put that:
https://github.com/janeczku/calibre-web/pull/1418